hs/src/Reach/Eval.hs

```diff
@@ -80,9 +80,208 @@ compileDApp shared_lifts exports (SLV_Prim (SLPrim_App_Delay at top_s (top_env,
   let sps_ies = ar_pie
   let sps_apis = ar_isAPI
   let sps = SLParts {..}
-  return $ DLProg at final_dlo' sps dli exports ar_views ar_apis ar_events final
+  final' <- pan final
+  return $ DLProg at final_dlo' sps dli exports ar_views ar_apis ar_events final'
 compileDApp _ _ _ = impossible "compileDApp called without a Reach.App"
 
+class Pandemic a where
+  pan :: a -> App a
+
+instance Pandemic a => Pandemic [a] where
+  pan = mapM pan
+
+instance Pandemic a => Pandemic (Maybe a) where
+  pan = mapM pan
+
+instance (Pandemic b) => Pandemic (Either a b) where
+  pan = mapM pan
+
+instance Pandemic DLStmts where
+  pan = mapM pan
+
+instance Pandemic DLExpr where
+  pan = \case
+    DLE_Arg at arg -> DLE_Arg at <$> pan arg
+    DLE_LArg at larg  -> DLE_LArg at <$> pan larg
+    DLE_Impossible at i err -> return $ DLE_Impossible at i err
+    DLE_VerifyMuldiv at cxt ct argl err -> do
+      r <- pan argl
+      return $ DLE_VerifyMuldiv at cxt ct r err
+    DLE_PrimOp at primop argl -> DLE_PrimOp at primop <$> pan argl
+    DLE_ArrayRef at arg1 arg2 -> DLE_ArrayRef at <$> pan arg1 <*> pan arg2
+    DLE_ArraySet at arg1 arg2 arg3 -> DLE_ArraySet at <$> pan arg1 <*> pan arg2 <*> pan arg3
```

Comment:
Btw, there are "standard names" for all the fields near their definition, in the Pretty instance

Reply:
i don't see a Pretty instance for DLExpr